### PR TITLE
chore: ScreenNames updates

### PR DESCRIPTION
**Pull request by calebpanza**
## ✏️ Description
In an effort to better standardize our screens and navigation, this PR introduces a `ScreenNames` object that gets exported from the `screens` directory. It contains all the navigation names for every screen exported from this directory.

Tests written make sure that every screen that is exported has a name associated with it and that the names match. This is for readability and standardization.

## 🔬 To Test
1. Read through the code
2. Run `yarn test screens`

## 📝 Checklist
- [x] I have performed a self-review of my code
- [x] I have tested my code works in the project (nothing breaks)
- [x] I have commented my code in hard-to-understand areas
- [x] New and existing tests and linting pass locally with my changes
- [x] I have made corresponding changes to the documentation (new envs?)

### CODE REVIEW

File: templates/mobile/jest.setup.js [hunk 1]
```diff
@@ -3,4 +3,15 @@ import FRAGMENTS from '@apollosproject/ui-fragments';
 
 require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
 
+jest.mock('react-native-image-crop-picker', () => {
```
Comment (redreceipt) [inline templates/mobile/jest.setup.js:6]: wanna throw this in the `__mocks__` folder to match our current mock setup?

File: templates/mobile/src/screens/ScreenNames.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export default {
+  Communities: 'Communities',
+  CommunitySingle: 'CommunitySingle',
+  //   UserPostInput: 'UserPostInput',
```
[the remark this answers is not in the document]
Comment (calebpanza) [reply templates/mobile/src/screens/ScreenNames.js:4]: Meant to remove that! Lol. There's a followup PR that adds that screen. If I add it back, tests will break.

File: templates/mobile/jest.setup.js (hunk 1 above)
Comment (calebpanza) [reply templates/mobile/jest.setup.js:6]: Sure thing!

File: templates/mobile/src/index.js [hunk 3]
```diff
@@ -253,23 +253,23 @@ const App = () => {
...
+                                name={ScreenNames.CommunitySingle}
+                                component={CommunitySingle}
                                 options={{
                                   ...TransitionPresets.SlideFromRightIOS,
                                   stackPresentation: 'push',
                                 }}
                               />
                               <Screen
-                                name="Post"
+                                name={ScreenNames.UserPostInput}
```
Comment (redreceipt) [inline templates/mobile/src/index.js:264]: so now isn't this undefined?

**Review by redreceipt — CHANGES_REQUESTED** [review]
move mock into the `__mocks__` folder

File: templates/mobile/src/index.js (hunk 3 above)
Comment (calebpanza) [reply templates/mobile/src/index.js]: Fragment brought over by the follow up PR. Good catch

File: templates/mobile/__mocks__/react-native-image-crop-picker.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default jest.fn();
```
Comment (redreceipt) [inline templates/mobile/__mocks__/react-native-image-crop-picker.js:1]:
```suggestion
export default "ImagePicker";
```

Comment (redreceipt) [reply templates/mobile/__mocks__/react-native-image-crop-picker.js:1]: this will look better in snapshots :)

**Review by redreceipt — APPROVED** [review]

**Pull request merged**